### PR TITLE
Add configuration profiles with .kpf format (Feature 3c)

### PR DESCRIPTION
**Pull request by ikari-pl**
## Summary
- Implement named configuration profiles for saving/loading emulator settings
- New commands: `profile list/current/load/save/delete`
- Simple INI-style .kpf format (human-readable, comment support)
- 4 built-in profiles: cpc464, cpc664, cpc6128, 6128plus
- Profile storage in `~/.koncepcja/profiles/`
- Name validation and built-in profile protection
- 11 new unit tests (338 total pass)

## New files
- `src/config_profile.h` / `src/config_profile.cpp` — ConfigProfileManager with INI serialization
- `test/config_profile.cpp` — 11 tests including round-trip, built-in listing, edge cases

## Test plan
- [x] Build passes on all platforms
- [x] 11 new unit tests pass
- [x] Save/load round-trip preserves all 11 settings
- [x] Built-in profiles always present in list
- [x] Invalid names rejected (path traversal, empty, etc.)
- [x] Built-in profiles cannot be deleted

### CODE REVIEW

File: src/config_profile.cpp [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+void ConfigProfileManager::set_profile_dir(const std::string& dir) {
+    profile_dir_ = dir;
+}
+
+std::string ConfigProfileManager::profile_dir() const {
+    if (!profile_dir_.empty()) return profile_dir_;
+    const char* home = getenv("HOME");
+    if (!home) home = getenv("USERPROFILE");
+    if (!home) return ".koncepcja/profiles";
+    return std::string(home) + "/.koncepcja/profiles";
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/config_profile.cpp]: Fixed. Now using fs::path with / operator for portable path construction.

File: src/config_profile.cpp [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+}
+
+std::string ConfigProfileManager::profile_path(const std::string& name) const {
+    return profile_dir() + "/" + name + ".kpf";
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/config_profile.cpp]: Fixed. Same fs::path approach applied here.
